src/angular-flash.js

```diff
@@ -1,134 +1,208 @@
-                Flash.setShowClose(scope.showClose);
-                function onDismiss(flash) {
-                    if (typeof scope.onDismiss !== 'function') return;
-                    scope.onDismiss({flash: flash});
-                }
+app.run(['$rootScope', function ($rootScope) {
+    return $rootScope.flashes = {};
+}]);
 
-                Flash.setOnDismiss(onDismiss);
+app.directive('dynamic', ['$compile', function ($compile) {
+    return {
+        restrict: 'A',
+        replace: true,
+        link: function link(scope, ele, attrs) {
+            return scope.$watch(attrs.dynamic, function (html) {
+                ele.html(html);
+                return $compile(ele.contents())(scope);
+            });
+        }
+    };
+}]);
+
+app.directive('closeFlash', ['$compile', '$rootScope', 'Flash', function ($compile, $rootScope, Flash) {
+    return {
+        link: function link(scope, ele, attrs) {
+            return ele.on('click', function () {
+                var id = parseInt(attrs.closeFlash, 10);
+                Flash.dismiss(id);
+                $rootScope.$apply();
+            });
+        }
+    };
+}]);
+
+app.directive('flashMessage', ['Flash', function (Flash) {
+    return {
+        restrict: 'E',
+        scope: {
+            duration: '=',
+            showClose: '=',
+            flashScope: '@?',
+            onDismiss: '&'
+        },
+        template: '<div role="alert" ng-repeat="flash in $root.flashes[flashScope] track by $index" id="{{flash.config.id}}" class="alert {{flash.config.class}} alert-{{flash.type}} alert-dismissible alertIn alertOut"><div type="button" class="close" ng-show="flash.showClose" close-flash="{{flash.id}}"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></div> <span dynamic="flash.text"></span> </div>',
+        link: function link(scope, ele, attrs) {
+            scope.flashScope = angular.isUndefined(scope.flashScope) || scope.flashScope == defaultFlashScope ? defaultFlashScope : scope.flashScope;
+            var flashFact = Flash.getScope(scope.flashScope);
+            flashFact.setDefaultTimeout(scope.duration);
+            flashFact.setShowClose(scope.showClose);
+            function onDismiss(flash) {
+                if (typeof scope.onDismiss !== 'function') return;
+                scope.onDismiss({ flash: flash });
             }
-        };
-    }
-]);
 
-app.factory('Flash', [
-    '$rootScope', '$timeout', function($rootScope, $timeout) {
-        const dataFactory = {};
-        let counter = 0;
-        dataFactory.setDefaultTimeout = function(timeout) {
+            flashFact.setOnDismiss(onDismiss);
+        }
+    };
+}]);
+
+app.factory('Flash', ['$rootScope', '$timeout', function ($rootScope, $timeout) {
+    var scopedDataFactory = {
+        scopes: {}
+    };
+    var counter = 0;
+
+    scopedDataFactory.getScope = function getFlashScope(flashScope) {
+        if(!flashScope) flashScope = defaultFlashScope;
```

Comment:
It is es2015 js. You can use
`scopedDataFactory.getScope = function getFlashScope(flashScope = defaultFlashScope) {`